server/supernova/server/server.hpp

```diff
@@ -226,18 +228,16 @@ class nova_server : public asynchronous_log_thread,
 
 inline void run_scheduler_tick(void) {
     const int blocksize = sc_factory->world.mBufLength;
-    const int input_channels = sc_factory->world.mNumInputs;
-    const int output_channels = sc_factory->world.mNumOutputs;
```

Comment:
to me it's a bad sign that the server and sc_factory (which is a unique_ptr to SC_World) have different ideas of the number of ins and outs. can you delay instantiating sc_factory until that's known, in nova_server::prepare_backend()? or otherwise, perhaps modify these values during audio backend config?